Lib/test/libregrtest/main.py

```diff
@@ -628,7 +628,12 @@ def create_temp_dir(self):
         # Define a writable temp dir that will be used as cwd while running
         # the tests. The name of the dir includes the pid to allow parallel
         # testing (see the -j option).
-        pid = os.getpid()
+        # Emscripten and WASI have stubbed getpid(), Emscripten has only
+        # milisecond clock resolution. Use time_ns() + randint() instead.
+        if sys.platform in {"emscripten", "wasi"}:
+            pid = time.time_ns() + random.randint(0, 1000000)
```

Comment:
Would you mind to rename pid to none?

``time.time_ns() + random.randint(0, 1000000)`` creates around 19 digits. Maybe the randint() call is enough?

Also, I suggest writing ``1_000_000`` to make the number more readable ;-)